src/Components/apiConnect.js

```diff
@@ -0,0 +1,53 @@
+class BookstoreAPI {
+  static baseURL = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/2iHiTbOoSnfgGXhW4oy3/books/';
+
+  static getBooks = async () => fetch(this.baseURL, {
+    method: 'GET',
+    headers: {
+      'Content-type': 'application/json; charset=UTF-8',
+    },
+  }).then(async (response) => {
+    if (response.ok) {
+      try {
+        const data = await response.json();
+        const books = [];
+        Object.entries(data).forEach(
+          ([id, book]) => books.push({
+            item_id: id,
+            title: book[0].title,
+            author: book[0].author,
+            category: book[0].category,
+          }),
+        );
+        return books;
+      } catch (e) {
+        return [];
+      }
+    }
+    return null;
+  });
+
+  static addBook = async ({
+    id, bookName, authorName, category,
+  }) => fetch(this.baseURL, {
+    method: 'POST',
+    body: JSON.stringify({
+      item_id: id,
```

Comment:
- [ ] Make sure that you implement well the method addBook. Add book features should work the same as before adding the API. means by clicking on the addbook button, this method should dispatch the existing `ADD_BOOK`. For now when we click on the button to add a new book this error appears:
![image](https://user-images.githubusercontent.com/71874425/174171410-aa0501e2-bad3-4051-9e79-330907911d17.png)
- Kindly, Fix this issue.

Reply:
Can you help me with it because 
I spent more than four hours on it
I still have the same problem 
Please!